tensorflow_io/core/kernels/image_jpeg2k_kernels.cc

```diff
@@ -0,0 +1,233 @@
+
+    if (p->offset_ + p_nb_bytes < p->length_) {
+      p->offset_ += p_nb_bytes;
+      return p_nb_bytes;
+    }
+    p->offset_ = p->length_;
+    return (OPJ_OFF_T)-1;
+  }
+
+  static OPJ_BOOL SeekFn(OPJ_OFF_T p_nb_bytes, void* p_user_data) {
+    OpjStreamCallback* p = static_cast<OpjStreamCallback*>(p_user_data);
+    if (p_nb_bytes < p->length_) {
+      p->offset_ = p_nb_bytes;
+      return OPJ_TRUE;
+    }
+    p->offset_ = p->length_;
+    return OPJ_FALSE;
+  }
+
+  static void OpjStreamFreeUserDataFn(void* p_user_data) {}
+};
+
+class DecodeJPEG2K : public OpKernel {
+ public:
+  explicit DecodeJPEG2K(OpKernelConstruction* context) : OpKernel(context) {}
+
+  void Compute(OpKernelContext* context) override {
+    const Tensor& contents_tensor = context->input(0);
+    OP_REQUIRES(context, TensorShapeUtils::IsScalar(contents_tensor.shape()),
+                errors::InvalidArgument("contents must be scalar, got shape ",
+                                        contents_tensor.shape().DebugString()));
+    auto contents = contents_tensor.scalar<tstring>()();
+
+    OPJ_CODEC_FORMAT format = OPJ_CODEC_JP2;
+
+    std::unique_ptr<opj_image_t, void (*)(opj_image_t*)> l_image(
+        nullptr, [](opj_image_t* p) {
+          if (p != nullptr) {
+            opj_image_destroy(p);
+          }
+        });
+    std::unique_ptr<opj_codec_t, void (*)(opj_codec_t*)> l_codec(
+        opj_create_decompress(format), [](opj_codec_t* p) {
+          if (p != nullptr) {
+            opj_destroy_codec(p);
+          }
+        });
+
+    OpjMsgCallback msg;
+    opj_set_info_handler(l_codec.get(), OpjMsgCallback::InfoCallback, &msg);
+    opj_set_warning_handler(l_codec.get(), OpjMsgCallback::WarningCallback,
+                            &msg);
+    opj_set_error_handler(l_codec.get(), OpjMsgCallback::ErrorCallback, &msg);
+
+    std::unique_ptr<opj_stream_t, void (*)(opj_stream_t*)> l_stream(
+        opj_stream_default_create(OPJ_TRUE), [](opj_stream_t* p) {
+          if (p != nullptr) {
+            opj_stream_destroy(p);
+          }
+        });
+    OP_REQUIRES(context, (l_stream.get() != nullptr),
+                errors::InvalidArgument("unable to create stream"));
+
+    OpjStreamCallback data(contents.data(), contents.size());
+
+    opj_stream_set_user_data(l_stream.get(), &data,
+                             OpjStreamCallback::OpjStreamFreeUserDataFn);
+    opj_stream_set_user_data_length(l_stream.get(), contents.size());
+    opj_stream_set_read_function(l_stream.get(), OpjStreamCallback::ReadFn);
+    opj_stream_set_skip_function(l_stream.get(), OpjStreamCallback::SkipFn);
+    opj_stream_set_seek_function(l_stream.get(), OpjStreamCallback::SeekFn);
+
+    opj_dparameters_t l_param;
+    opj_set_default_decoder_parameters(&l_param);
+
+    // do not use layer decoding limitations
+    // l_param.cp_layer = 0;
+
+    // do not use resolutions reductions
+    // l_param.cp_reduce = 0;
```

Comment:
Remove commented code?

Reply:
Still here?

Reply:
@terrytangyuan I added a TODO, as those two parameters might be needed. Haven't find a sample to test yet, so leave them in TODO comment.